app/services/payment_providers/moneyhash/payments/create_service.rb

```diff
@@ -0,0 +1,31 @@
+# frozen_string_literal: true
+
+module PaymentProviders
+  module Moneyhash
+    module Payments
+      class CreateService < BaseService
+        include ::Customers::PaymentProviderFinder
+
+        def initialize(payment:, reference:, metadata:)
+          @payment = payment
+          @reference = reference
+          @metadata = metadata
+          @invoice = payment.payable
+          @provider_customer = payment.payment_provider_customer
+
+          super
+        end
+
+        def call
+          PaymentRequests::Payments::MoneyhashService.new(@invoice).create
```

Comment:
I'd expect the opposite. The `PaymentRequests::Payments:: MoneyhashService` service to rely on the `PaymentProviders::Moneyhash::Payments::CreateService` to create the payment and then do specific things (like sending the email).

I think all the attributes are leftovers too.